assets/src/util/assignment.js

```diff
@@ -145,14 +166,34 @@ const sumAssignmentGoalGrade = assignments => sum(
   assignments.map(a => a.goalGrade || 0)
 )
 
-const sortAssignmentsByWeek = assignments => {
-  const assignmentsWithDueDates = assignments
+// assignmentGroup.name can be null
+const sortByAssignmentGroupName = (assignments) => {
+  return assignments.sort((a, b) => {
+    return (a.assignmentGroup.name === null && b.assignmentGroup.name === null) ? 0
+      : a.assignmentGroup.name ? a.assignmentGroup.name.localeCompare(b.assignmentGroup.name)
```

Comment:
Are you trying to sort alphabetically here? 

Could this possibly be simplified to: 

```js
const sortByAssignmentGroupName = assignments => assignments.sort((a, b) => {
  if (a.assignmentGroup.name === null || b.assignmentGroup.name === null) return 0
  if (a.assignmentGroup.name < b.assignmentGroup.name) return -1
  if (a.assignmentGroup.name > b.assignmentGroup.name) return 1
  else return 0
})
```

Reply:
Would that not fail to correctly sort elements reliably when one of the assignmentGroup.name values is null?

Reply:
typically assignment's will always be in a group, if that is not the case you are seeing then atleast we can reduce one return statement like this `=> assignments.sort((a, b) => {` and keep the rest as if and else logic same

Reply:
> Would that not fail to correctly sort elements reliably when one of the assignmentGroup.name values is null?

I think I need to test that case to double check, but I think if either one or both of the `assignmentGroup.names` is null, it won't sort that pair.

Reply:
Yeah that is what I meant by unreliably.  They will retain the order they were presented in, so if they were swapped as input into different sorts they would come out in a different order.

Reply:
What I meant was that if the 2 inputs were in a different order on two different sorts, if it just returned 0 if either assignmentGroup,name was null, they 2 inputs would be in a different order after the sort was applied so 2 arrays with identical contents in a different order could be sorted with different outcomes.

Reply:
> typically assignment's will always be in a group, if that is not the case you are seeing then atleast we can reduce one return statement like this `=> assignments.sort((a, b) => {` and keep the rest as if and else logic same

assignmentgroup cannot be null but  assignmentgroup.name can be.